Formula/mpi4py.rb

```diff
@@ -0,0 +1,30 @@
+class Mpi4py < Formula
+  desc "Python bindings for MPI"
+  homepage "https://mpi4py.readthedocs.io"
+  url "https://bitbucket.org/mpi4py/mpi4py/downloads/mpi4py-3.0.2.tar.gz"
+  sha256 "f8d629d1e3e3b7b89cb99d0e3bc5505e76cc42089829807950d5c56606ed48e0"
+  head "https://bitbucket.org/mpi4py/mpi4py.git", :branch => "master"
+
+  depends_on "cython" => :build
+  depends_on "open-mpi"
+  depends_on "python"
+
+  def install
+    system "#{Formula["python"].opt_bin}/python3",
+           *Language::Python.setup_install_args(libexec)
+
+    system "python3", "setup.py",
+      "build", "--mpicc=mpicc -shared", "--parallel=#{ENV.make_jobs}",
+      "install", "--prefix=#{prefix}",
+      "--single-version-externally-managed", "--record=installed.txt"
+  end
+
+  test do
+    system "#{Formula["python"].opt_bin}/python3",
+           "-c", "import mpi4py"
+    system "#{Formula["python"].opt_bin}/python3",
+           "-c", "import mpi4py.MPI"
+    system "#{Formula["python"].opt_bin}/python3",
+           "-c", "import mpi4py.futures"
```

Comment:
Can you please improve the tests here a little bit? Just importing the module may not be sufficient. Using it (briefly) would be nice. Look at numpy's test for ideas.

Reply:
If importing is not enough, I suggest the following quick testing, both of them (even the `helloworld` one) exercise basic MPI point-to-point communication. Testing beyond that may prove unreliable, trust me, do not open a can or worms...

```
$ mpiexec -n 4 python3 -m mpi4py.run -m mpi4py.bench helloworld
Hello, World! I am process 0 of 4 on localhost.
Hello, World! I am process 1 of 4 on localhost.
Hello, World! I am process 2 of 4 on localhost.
Hello, World! I am process 3 of 4 on localhost.

$ mpiexec -n 4 python3 -m mpi4py.run -m mpi4py.bench ringtest -l 10 -n 1024
time for 10 loops = 5.55515e-05 seconds (4 processes, 1024 bytes)
```

Reply:
I agree, we don't need extensive testing, just confirmation that mpi4py is able to use Homebrew's open-mpi

Reply:
btw, how flexible is this package regarding OpenMPI/MPICH and is there a reason for the current preference?

Reply:
Not 100% sure what you mean. Homebrew defaults to Open-MPI only so that all MPI packages will play nice together. I started to put together MPICH tooling over at the unofficial brew-sci tap but haven’t made much progress.

Reply:
Not 100% sure what you mean. Homebrew defaults to Open-MPI only so that all MPI packages will play nice together. I started to put together MPICH tooling over at the unofficial brew-sci tap but haven’t made much progress.